deploy/tasks/pre-build.yaml

```diff
@@ -0,0 +1,145 @@
+  params:
+    - name: IMAGE_URL
+      description: URL of the OCI image to use.
+      type: string
+    - name: NAME
+      description: Name of the pipeline run (i.e. unique dependency build name)
+      type: string
+    - name: GIT_SCRIPT
+      description: Git clone commands
+      type: string
+    - name: GIT_IDENTITY
+      description: Git username
+      type: string
+    - name: GIT_URL
+      description: URL to determine whether we're using gitlab or github
+      type: string
+    - name: GIT_DEPLOY_TOKEN
+      description: Name of jvm-build-git-repo-secrets secret containing git password/token to use.
+      type: string
+    - name: GIT_SSL_VERIFICATION
+      description: Whether to disable ssl verification
+      type: string
+      default: "false"
+    - name: GIT_REUSE_REPOSITORY
+      description: Whether to reuse existing git repository or create new one
+      type: string
+    - name: SCM_URL
+      description: Reference to the git repository
+      type: string
+    - name: SCM_HASH
+      description: Git hash
+      type: string
+    - name: RECIPE_IMAGE
+      description: The image from the build recipe to use
+    - name: BUILD_SCRIPT
+      description: The build script to embed with the Containerfile
+    - name: PREPROCESSOR_ARGS
+      description: The arguments for the build preprocessor
+    - name: ORAS_OPTIONS
+      type: string
+      description: Optional environment variable string for build-trusted-artifacts
+      default: ""
+    - name: JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE
+      description: Name of the processor image. Useful to override for development.
+      type: string
+      default: "quay.io/redhat-appstudio/hacbs-jvm-build-request-processor:dev"
+  results:
+    - name: PRE_BUILD_IMAGE_DIGEST
+      description: Digest of the image just built
+    - name: GIT_ARCHIVE
+      description: Git archive information
+  workspaces:
+    - description: The git repo will be cloned onto the volume backing this Workspace.
+      name: source
+      mountPath: /var/workdir
+    - name: tls
+  steps:
+    # Should we use our own git clone task? Or embed (somehow) Konflux's version?
+    - name: git-clone
+      image: $(params.RECIPE_IMAGE)
+      computeResources:
+        limits:
+          cpu: 300m
+          memory: 512Mi
+        requests:
+          cpu: 10m
+          memory: 512Mi
+      securityContext:
+        runAsUser: 0
+      env:
+        - name: GIT_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: jvm-build-git-secrets
+              key: .git-credentials
+      script: |
+        $(params.GIT_SCRIPT)
+    - name: preprocessor
+      image: $(params.JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE)
+      securityContext:
+        runAsUser: 0
+      computeResources:
+        limits:
+          cpu: 300m
+          memory: 512Mi
+        requests:
+          cpu: 10m
+          memory: 512Mi
+      script: |
+        $(params.BUILD_SCRIPT)
```

Comment:
To embed the Containerfile / build-script into the repository **currently** passing them through from the operator (which is generating them). Eventually I think the preprocessor should generate them (for the following steps (`create-pre-build-source` and `create-pre-build-image`) to use).

Reply:
Yes I believe it would be cleaner that the preprocessor generates the build script and propagates it as a result to following steps.